### DISCUSSION TITLE
Problem when run nmake on VC-WIN64A configuration for version 3.4.0

### DISCUSSION THREAD

**Discussion by benjamin-xu-abb** [openssl/openssl#26167] [Q&A] [answered]
When I tried to build openssl-3.4.0 for VC-WIN64A, I had a problem as shown in attached nmake.png.

My system is a VM running Windows Server 2019 Standard with Visual Studio 2022 version 17.12.3 installed. 

Please let me know if you need more information.

Thanks.

![configure](https://github.com/user-attachments/assets/74e7f1d5-c3ae-4651-b578-0bb436a74999)
![nmake](https://github.com/user-attachments/assets/c5d4e780-df10-44cf-8c8a-b46e916fb730)
[dump.txt](https://github.com/user-attachments/files/18114834/dump.txt)

**Comment by nhorman** [comment, marked as answer]
It looks like you're trying to build openssl for 64 bit using a visual studio environment setup to build 32 bit binaries, based on this line in your png above:
```
fatal error LNK1112: module machine type `x86` conflicts with target machine type 'x64'
```

Try, in your vs studio command line window, running the following command:
```
vcvarsall.bat x86
```
Note, vcvarsall.bat is a visual studio executable, and it may not be in your path, so you may need to specify a full path to the binary

**Comment by benjamin-xu-abb** [comment]
Thanks for the fast reply. 

Tried your recommendation and it works, except used 'x64' instead of 'x86'.  Now the nmake succussed and nmake test passed.

Really appreciate your help.
